architecture/protocol.go

```diff
@@ -147,6 +147,16 @@ func (command *Command) Parse(rawCommand string) (bool, error) {
 			return true, command.Err
 		}
 
+		// Check if parameter types are correct
+		for i, _ := range opts.Params {
+			if opts.ParamInteger[i] {
+				_, err := strconv.Atoi(parts[i+1])
```

Comment:
@marekyggdrasil The `if opts.ParamInteger[i]` logic would be better placed in the loop on line 164 so that opts.Params is only ranged over once.  And if there's an error, immediately return after setting `command.Err`--so after whatever line 155 becomes after it's moved into the other `opts.Params` loop.

Also, I apologize for suggesting `strconv.Atoi`--it returns an `int` when we need an `int64`, so `strconv.ParseInt` is the only viable option.